### ISSUE TITLE
Dark mode / night theme

### ISSUE THREAD

**Issue by BasThomas** [rnystrom/Freetime#198]
Would love this. Definitely not too important though.

**Comment by shaps80** [comment]
Haha I wondered how long it would take for someone to request this.

**Comment by BasThomas** [comment]
This could go together with a dark app icon as well

**Comment by Sherlouk** [comment]
@Huddie @BrianLitwin @rnystrom 

Anybody got a mockup design or palette they think would work nice as a dark mode? Need some inspiration before I go rainbow city on the app :trollface:

**Comment by kocheck** [comment]
@Sherlouk This is a huge interest of mine 👍 I might spin something up this weekend to play with.

**Comment by Sherlouk** [comment]
I spent a week turning my Slack into dark mode for a reason 😂Dark mode should be a requirement for all apps :trollface: 

Can't wait to see what you make!

**Comment by rnystrom** [comment]
Very very interested. The hardest part for us will be text colors. Might have to update StyledTextKit to support themes or something so we can regen attributed strings without rebuilding.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ggliv** [comment]
I understand that this is an ancient thread but what’s the status on this? I’d love to see this happen.

**Comment by BasThomas** [comment]
Hey @ggliv, thanks for bringing it up! Let's wait and see what Apple announces on Monday... and see where to go from there.

There's also still a work-in-progress pull request for a dark mode [here](https://github.com/GitHawkApp/GitHawk/pull/2553).

**Comment by Huddie** [comment]
Seems like dark mode will kinda come for free with iOS 13. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jdisho** [comment]
I guess, we still need to support dark mode for older iOS versions, right? 🤔

**Comment by Sherlouk** [comment]
Didn't take you folks long to start updating this 😅 Personally I don't think we should actively try to support dark mode on iOS 12-

Dark mode exclusive to iOS 13+ using OS systems sounds like a simpler solution and less maintenance longer term.

**Comment by iDjay** [comment]
Would love to see one too :)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ideologysec** [comment]
Definitely throwing my vote in for a dark mode in iOS 13+ (previous versions not so much; let's aim for the MVP of getting it implemented, and if there's enough demand it can maybe be backported...)

**Comment by ggliv** [comment]
@BasThomas Any news?

**Comment by BasThomas** [comment]
Hi! I'd love to see this, but am waiting for an update [on this issue](https://github.com/GitHawkApp/GitHawk/issues/2806).

In the meantime, anyone's free to start on this!

**Comment by flowinho** [comment]
> Seems like dark mode will kinda come for free with iOS 13.

Only for UI that is using SwiftUI. As this is not applicable for GitHawk, it will not automatically transition to supporting DarkMode.

> Definitely throwing my vote in for a dark mode in iOS 13+ (previous versions not so much; let's aim for the MVP of getting it implemented, and if there's enough demand it can maybe be backported...)

This would mean re-implementing everything either in SwiftUI (which is imho not yet production ready) or implementing it in one of the old-fashioned ways.

> Dark mode exclusive to iOS 13+ using OS systems sounds like a simpler solution and less maintenance longer term.

I definitely second this. However this would require a significant amout of refactoring.

**Comment by ideologysec** [comment]
> Only for UI that is using SwiftUI. As this is not applicable for GitHawk, it will not automatically transition to supporting DarkMode.

Uhhh.... no. SwiftUI automatically supports Dark Mode, but it is in no way a requirement for it. Have you read the API and design docs? (https://developer.apple.com/documentation/appkit/supporting_dark_mode_in_your_interface)

@flowinho  I'd respond to the rest of your quotes but it's as wrong as this - seems like your post is based on the WWDC keynote and extrapolations?

Yes, there's more work required to create a dark theme, but it doesn't require a reimplementation in SwiftUI, and it's a little easier than "the old-fashioned ways."

**Comment by ijm8710** [comment]
This is above my head but just be aware of #2553; not sure if the WIP helps at all, especially since I think that was done before it was announced that dark mode would be more natively supported, but just didn’t want it to fall completely on the wayside if it did have value

**Comment by kocheck** [comment]
I have been deep into the dark mode docs for another design project this past month. I think I am going to take a crack at using the default apple dark mode stuff here if you all are still interested. I know a lot of refactoring is required.

**Comment by kocheck** [comment]
I started working out the color styles tonight, trying to use as much of what apple gives us as possible. Looks like the designs files (overall) are a little behind production and could use some love  ❤️. 

I should have the color guide done first. Following the screen designs. I think I am going to recreate the screens from what I can see in production right now. 

**Light and Dark Mode colors:**

![image](https://user-images.githubusercontent.com/7952000/63397399-6ba60b00-c398-11e9-9779-945c4c2c7ae5.png)

Going to keep chipping away at this 👍

**Comment by raygervais** [comment]
Any update on this @kocheck ?

**Comment by Huddie** [comment]
I think the GitHub Mobile app takes care of this issue 😀

**Comment by raygervais** [comment]
You’re not wrong haha! Still, diversity in the app space doesn’t hurt anyone. Would be happy to try my hand at this (very new to iOS Open Source) still.

**Comment by BasThomas** [comment]
Go for it 😊
